examples/Nat/NATBase.cs

```diff
@@ -0,0 +1,287 @@
+using PacketDotNet;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Text;
+
+namespace Pax.Examples.Nat
+{
+  /// <summary>
+  /// Base implementation of a protocol-specific NAT.
+  /// </summary>
+  /// <typeparam name="TPacket">The type of packet.</typeparam>
+  /// <typeparam name="TNode">The type of node.</typeparam>
+  /// <typeparam name="TEncapsulation">The type of the packet encapsulation.</typeparam>
+  public abstract class NATBase<TPacket,TNode,TEncapsulation> where TPacket : Packet where TNode : Node where TEncapsulation : PacketEncapsulation<TPacket,TNode>
```

Comment:
I think it's very elegant how this class factors out the NAT behaviour that can be instantiated by different protocols.